src/node_perf.cc

```diff
@@ -104,8 +104,8 @@ void Mark(const FunctionCallbackInfo<Value>& args) {
   auto marks = env->performance_marks();
   (*marks)[*name] = now;
 
-  // TODO(jasnell): Once Tracing API is fully implemented, this should
-  // record a trace event also.
+  TRACE_EVENT_COPY_MARK_WITH_TIMESTAMP(
+      "node.perf,node.perf.usertiming", *name, now / 1e3);
```

Comment:
IIRC, this is doing a floating point divide which is more expensive than an integer divide.

Reply:
ah yeah, good point :-)

Reply:
Confirmed. Floating point divide + 2 conversions between integer and floating point.

```c++
#include <stdint.h>
uint64_t foo(uint64_t nanos) {
   return nanos / 1e3;
}
// clang -O3 -mllvm --x86-asm-syntax=intel -S foo.c
```

```asm
	movq	xmm0, rdi
	punpckldq	xmm0, xmmword ptr [rip + LCPI0_0] ## xmm0 = xmm0[0],mem[0],xmm0[1],mem[1]
	subpd	xmm0, xmmword ptr [rip + LCPI0_1]
	haddpd	xmm0, xmm0
	divsd	xmm0, qword ptr [rip + LCPI0_2]
	movsd	xmm1, qword ptr [rip + LCPI0_3] ## xmm1 = mem[0],zero
	movapd	xmm2, xmm0
	subsd	xmm2, xmm1
	cvttsd2si	rax, xmm2
	movabs	rcx, -9223372036854775808
	xor	rcx, rax
	cvttsd2si	rax, xmm0
	ucomisd	xmm0, xmm1
	cmovae	rax, rcx
	pop	rbp
	ret
```